README.md

```diff
@@ -1,3 +1,78 @@
-# Tracking ML
+# Tracking MLflow jobs on Openshift
 
-Demo of tracking machine learning metrics into mlflow running on k8s
+This is a demo application which includes templates that can be used to run mlflow jobs on Openshift.
+
+## Running on OpenShift
+This is a two stage process, because you need to create the image only once and then you can create multiple containers using the same image but with different parameters.
+
+### Create a container image
+To run this application on openshift, we need to create a container image for it.  To do this we have a image build template [here in the repository](https://github.com/AICoE/experiment-tracking-template/blob/master/openshift/mlflow-experiment-image-buildconfig.yaml).
+
+This template will take a base image, and install all the dependencies listed in the Pipfile and store this new image to the image registry.
+
+To initiate a new image build, run the following command from the root of this repository:
+
+```
+oc new-app ./openshift/mlflow-experiment-image-buildconfig.yaml --param APPLICATION_NAME=mlflow-experiment-image-demo --param GIT_URI=https://github.com/AICoE/experiment-tracking-template.git --env APP_FILE=app.py
+```
+Here, the `APPLICATION_NAME` parameter is the name of the container image that will be built.
+<br/>You can set `GIT_URI` to the link for your repository to create a container image for your application.
+<br/>You can also change the `APP_FILE` env variable to your app file name.
+
+Bulding the container image can take a couple of minutes or more depending on the number of packages that need to be installed from the dependencies list.
+
+If the image build process has started you should see some output like this:
+```
+   * With parameters:
+   * APPLICATION_NAME=mlflow-experiment-image-demo
+   * GIT_URI=https://github.com/AICoE/experiment-tracking-template.git
+
+--> Creating resources ...
+imagestream.image.openshift.io "mlflow-experiment-image-demo" created
+buildconfig.build.openshift.io "mlflow-experiment-image-demo" created
+--> Success
+Build scheduled, use 'oc logs -f bc/mlflow-experiment-image-demo' to track its progress.
+Run 'oc status' to view your app.
+```
+### Deploy a mlflow job
+After the image is built, we can use it run a job on OpenShift.
+
+From the root of this repository, run the following command:
+```
+oc new-app ./openshift/mlflow-experiment-job-template.yaml --APPLICATION_NAME=mlflow-experiment-job-1 --param APP_IMAGE_NAME=mlflow-experiment-image-demo --param MLFLOW_TRACKING_URI=http://mlflow-server-url/ --env PARAM_ALPHA=0.4 --env PARAM_L1_RATIO=0.8
```

Comment:
same here, I guess `oc process` might be better. But I might be wrong...

Reply:
you cannot specify --env in `oc process` so I might switch back to `oc new-app`